modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala

```diff
@@ -155,11 +156,20 @@ object SharedOptionsUtil extends CommandHelpers {
       enableJmh: Boolean = false,
       jmhVersion: Option[String] = None,
       ignoreErrors: Boolean = false
-    ): bo.BuildOptions = {
-      val platformOpt =
-        if (js.js) Some(Platform.JS)
-        else if (native.native) Some(Platform.Native)
-        else None
+    ): Either[BuildException, bo.BuildOptions] = either {
+      val parsedPlatform = platform.map(Platform.normalize).flatMap(Platform.parse)
+      val platformOpt = value {
+        (parsedPlatform, js.js, native.native) match {
+          case (Some(p: Platform.JS.type), true, false)     => Right(Some(p))
```

Comment:
this and the fix below will let you get rid of `case (Some(p), _, _)`
```suggestion
          case (Some(p: Platform.JS.type), _, false)     => Right(Some(p))
```